components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/UpdateAdvancedConfigurations.java

```diff
@@ -59,9 +62,9 @@ private LocalAndOutboundAuthenticationConfig getLocalAndOutboundConfig(ServicePr
     private void updateCertificate(Certificate certificate, ServiceProvider serviceProvider) {
 
         if (certificate != null) {
-            if (certificate.getType() == Certificate.TypeEnum.PEM) {
+            if (certificate.getType().equals(TYPE_PEM)) {
```

Comment:
Define as `TYPE_PEM.equals(certificate.getType())` to avoid possible NPEs.

Reply:
Resolved